### PR TITLE
HEXA-483 S3 partial sync

### PR DESCRIPTION
**Pull request by ekatsah**

### CODE REVIEW

File: hexa/plugins/connector_s3/tests/test_views.py [hunk 1]
```diff
@@ -67,3 +69,45 @@ def test_object_detail_200_directory(self):
         self.assertIsInstance(response.context["object"], Object)
         self.assertIsInstance(response.context["object_card"], ObjectCard)
         self.assertIsInstance(response.context["object_grid"], ObjectGrid)
+
+    @mock_s3
+    @mock_sts
+    def test_bucket_refresh(self):
```
Comment (pvanliefland) [inline hexa/plugins/connector_s3/tests/test_views.py:75]: `test_object_refresh` :) #ocd

File: hexa/plugins/connector_s3/models.py [hunk 2]
```diff
@@ -455,25 +482,39 @@ def update_metadata(self, object_data):
...
             bucket=bucket,
             key=object_data["true_key"],
             parent_key=cls.compute_parent_key(object_data["true_key"]),
-            size=object_data["size"],
-            storage_class=object_data["StorageClass"],
-            type=object_data["type"],
+            storage_class=object_data.get("StorageClass", "STANDARD"),
             last_modified=object_data.get("LastModified"),
-            etag=object_data.get("ETag"),
+            etag=object_data["ETag"].strip('"') if "ETag" in object_data else None,
```
Comment (pvanliefland) [inline hexa/plugins/connector_s3/models.py:515]: Curious to hear about this - maybe worth a comment?

File: hexa/plugins/connector_s3/models.py [hunk 3]
```diff
@@ -455,25 +482,39 @@ def update_metadata(self, object_data):
 
         self.key = object_data["true_key"]
         self.parent_key = self.compute_parent_key(object_data["true_key"])
-        self.size = object_data["size"]
-        self.storage_class = object_data["StorageClass"]
+        if "size" in object_data:
+            self.size = object_data["size"]
+        elif "ContentLength" in object_data:
```
Comment (pvanliefland) [inline hexa/plugins/connector_s3/models.py:487]: `ContentLength` is for `head`?

**Review by pvanliefland — APPROVED** [review]
Nice work! Just a couple of questions but this can be merged.

File: hexa/plugins/connector_s3/models.py (hunk 2 above)
Comment (ekatsah) [reply hexa/plugins/connector_s3/models.py:515]: same as L184 it turns out. Except it isn't a bug in S3FS, it's the real API return. I didn't find any mention of this in docs.
Example: {
...
'ETag': '"195c1cb9a00c10b54d5a1000363fb0cb"',
...
}

File: hexa/plugins/connector_s3/models.py (hunk 3 above)
Comment (ekatsah) [reply hexa/plugins/connector_s3/models.py:487]: Yep

**Pull request merged**